### PR TITLE
feat(advisory-convergence): explain why sameHeadReroll.eligible is false

### LINKED ISSUE #1719: feat(advisory-convergence): explain why sameHeadReroll.eligible is false in report mode
## Background

`idd-advisory-convergence`'s report mode reports
`sameHeadReroll.eligible: false` with nothing in the output explaining
why, so an agent cannot self-diagnose a stuck AW6 reroll without
re-deriving the eligibility rule by hand from the instructions.

In `src/scripts/advisory-convergence.mts`, eligibility is a conjunction
of six conditions:

```ts
const sameHeadRerollEligible =
  !scopeNotApplicable &&
  !pending &&
  threadClause.satisfied &&
  dispositionEvidence.missingRegularCommentCount === 0 &&
  review.itemCount !== null &&
  review.itemCount > 0;
```

The emitted `sameHeadReroll` object carries only `eligible`, `count`,
`cap`, `exhausted`, `inFlight`, and `shouldRequest`. The report's
top-level `reasons` array is populated for convergence and waiver
outcomes only, and `dispositionEvidence` — whose
`missingRegularCommentCount` is one of the six terms — appears nowhere
in the returned object. `--help` offers no `--verbose` or `--debug`
flag to ask for more.

**Why this costs real time.** The reported instance had the primary
bot's review on current HEAD carrying `itemCount: 1` while every
visible GraphQL review thread was already `isResolved: true`. The
actual cause was a "Comments suppressed due to low confidence" item
embedded in the review's top-level **body text** rather than posted as
a review thread: it contributes to `itemCount` but is invisible to any
`reviewThreads` query, so it can never be resolved the normal way. That
root cause was found only by reading the raw review body. Nothing in
the CLI output pointed there, and the loop is autonomous — a phase that
cannot explain its own verdict is a phase that stalls or escalates
needlessly.

`docs/idd-helper-scripts.md`'s AW6 documentation already states that
`eligible` depends in part on
`dispositionEvidence.missingRegularCommentCount === 0`, so the field
the operator needs is documented as load-bearing while never being
emitted.

## Proposed change

1. Add an `ineligibleReasons` array to the `sameHeadReroll` object: one
   short, stable machine-readable token per failing term, empty when
   `eligible` is true. Derive it from the same six terms the
   conjunction already evaluates, so the array and the boolean cannot
   disagree — compute the terms once and build both from that result.
2. Emit the eligibility-relevant `dispositionEvidence` counters
   (at minimum `missingRegularCommentCount`) on the report object, so
   the numeric input is visible and not only its pass/fail verdict.
3. When the failing term is `review.itemCount > 0` while every visible
   thread is resolved, make the reason token or its accompanying text
   point at the review body as the place to look — this is the
   observed shape of a suppressed low-confidence item, and it is not
   otherwise discoverable from any thread query.
4. Update `schemas/advisory-convergence.schema.json` and its fixtures
   under `fixtures/schemas/` for the new fields.
5. Update `docs/idd-helper-scripts.md`'s AW6 section (and its
   `idd-template/` mirror) to document the new fields.
6. Add tests asserting that `ineligibleReasons` is empty exactly when
   `eligible` is true, and that each of the six terms produces its own
   token when it is the failing one.
7. Run `pnpm run build` so the generated `scripts/*.mjs` copies match.

Keeping `eligible` itself unchanged is deliberate: this issue makes the
existing verdict explicable, and does not alter which PRs qualify for a
same-HEAD reroll.

## Acceptance criteria

- [ ] `sameHeadReroll.ineligibleReasons` is present in report-mode
      output, is empty exactly when `sameHeadReroll.eligible` is true,
      and names each failing term when it is not.
- [ ] The report object exposes the `dispositionEvidence` counters
      that feed the eligibility conjunction.
- [ ] `schemas/advisory-convergence.schema.json` and the
      `fixtures/schemas/advisory-convergence.*.json` fixtures cover
      the new fields, and `node --test tests/*.test.mts` passes.
- [ ] Tests fail if a term is added to the eligibility conjunction
      without a corresponding reason token.
- [ ] `docs/idd-helper-scripts.md` and its `idd-template/` mirror
      document the new fields; `node scripts/audit-docs.mjs --check`
      passes.
- [ ] `pnpm run build:check` and `pnpm run lint` pass.

Refs: reported from an adopter repository stuck on an
`eligible: false` verdict whose root cause was a suppressed
low-confidence item in the review body.

---

_Autopilot suitability: 4 / 5 -- higher is more autopilot-suitable;
below the configured floor is human-oriented._

_Effort: M -- author-estimated size; a soft autopilot
selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
# Summary

`idd-advisory-convergence`'s report mode reported `sameHeadReroll.eligible: false`
with nothing in the output explaining why. This PR makes the verdict explain
itself without changing `eligible`'s value.

- Add `sameHeadReroll.ineligibleReasons`: one stable, machine-readable token
  per failing term of the six-term eligibility conjunction, empty exactly
  when `eligible` is `true`. All six named booleans and the token array are
  computed once and derived from the same array (`.every()` / `.filter().
  map()`), so the boolean and the array cannot disagree.
- Expose `dispositionEvidence.{missingRegularCommentCount,
  missingThreadCount}` on the report object, so the numeric input behind
  one of those six terms is visible directly, not only its pass/fail
  verdict.
- When `itemCount > 0` while every visible Copilot-authored thread is
  already resolved, extend the existing top-level `reasons` entry to point
  at the review body directly -- the observed shape of the reported adopter
  incident where a "Comments suppressed due to low confidence" item counted
  toward `itemCount` but never surfaced as a review thread, so no thread
  query could ever explain it.
- Update `schemas/advisory-convergence.schema.json` and its valid fixture,
  plus the schema/type reconciliation test, for the new fields.
- Document the new fields in AW6 (`docs/idd-helper-scripts.md` and its
  `idd-template/` mirror).
- Add tests: `ineligibleReasons` empty iff `eligible`, a pinned six-token
  set (so a term added to the conjunction without a paired token fails a
  test), one isolation test per term (two terms are documented as always
  co-firing, since an off-HEAD review always reports an unknown item
  count), and the review-body pointer text.

## Linked issue

Closes #1719

## Follow-up issues (if any)

- [x] none

## Background / rationale (if material)

`eligible` itself is unchanged -- this only makes the existing verdict
explicable. See the module header and inline comments in
`src/scripts/advisory-convergence.mts` for the full design rationale,
including why `review-item-count-not-positive` is deliberately written as
"unknown counts as satisfied" to keep it mutually exclusive with
`review-item-count-unknown` in the reported token list.

## IDD impact

- [ ] Instruction files changed
- [x] Template files changed
- [x] Helper scripts changed
- [ ] Config schema changed
- [ ] Security / credential / merge behavior changed

## Verification

- [x] `pnpm lint`
- [x] `pnpm test`
- [x] `node scripts/audit-docs.mjs --check`
- [x] `pnpm run docs:sync:check`
- [x] `node scripts/idd-doctor.mjs` (if applicable)

## Safety

- [ ] No secret-bearing examples
- [ ] No private repo names / URLs
- [ ] Merge policy impact reviewed
- [ ] Context budget impact reviewed

### CODE REVIEW

File: src/scripts/advisory-convergence.mts [hunk 1]
```diff
@@ -115,6 +115,23 @@ import {
 export const ADVISORY_CONVERGENCE_CHECK_SELECTOR =
   DEFAULT_ADVISORY_CONVERGENCE_CHECK_SELECTOR;
 
+/** #1719: stable, machine-readable tokens for `sameHeadReroll.-
+ * ineligibleReasons` -- one per boolean term of the `eligible` conjunction,
+ * in the same order the conjunction is written in
+ * `computeAdvisoryConvergenceVerdict` below, so a report-mode caller can
+ * self-diagnose a stuck AW6 reroll without re-deriving the eligibility rule
+ * from `idd-advisory-wait.instructions.md` by hand. Exported (rather than
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/advisory-convergence.mts]: **Accepted** — fixed in fc443d1aaaa32414bf6df312d400613dfd0db53a: the JSDoc code span was split across two comment lines (`SAME_HEAD_REROLL_INELIGIBLE_-` / `REASON`), rendering as an incorrect identifier. It is now a single, unbroken `SAME_HEAD_REROLL_INELIGIBLE_REASON` code span.

File: docs/idd-helper-scripts.md [hunk 2]
```diff
@@ -1542,21 +1554,50 @@ evidence, purely additively: `converged` / `waived` / `ready` are
...
+| `scope-not-applicable`                    | `applicability.status` is `not_applicable` (`advisoryWait.convergenceScope: "idd-claimed"` and this PR has no matching verified linked claim/branch).   |
+| `review-pending`                          | The primary bot has not yet reviewed current HEAD (`pending: true`). Always co-occurs with `review-item-count-unknown` below, since an off-HEAD review reports no usable item count. |
+| `unresolved-copilot-threads`              | `threads.satisfied` is `false` -- at least one Copilot-authored thread is neither resolved nor validly dispositioned.                                   |
+| `missing-regular-comment-disposition`     | `dispositionEvidence.missingRegularCommentCount` is non-zero -- an outstanding regular (non-thread) PR comment still lacks a fresh disposition marker.  |
+| `review-item-count-unknown`               | The latest review is on current HEAD but its comment count is unavailable (a GraphQL nullable-field edge case).                                         |
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply docs/idd-helper-scripts.md]: **Accepted** — fixed in fc443d1aaaa32414bf6df312d400613dfd0db53a: corrected the `review-item-count-unknown` row to also describe the off-HEAD case (co-fires with `review-pending`, since `resolveLatestCopilotReviewClause` reports `itemCount: null` for any non-matching-HEAD review), not only the on-HEAD nullable-count edge case.

File: idd-template/docs/idd-helper-scripts.md [hunk 3]
```diff
@@ -1542,21 +1554,50 @@ evidence, purely additively: `converged` / `waived` / `ready` are
...
+| `scope-not-applicable`                    | `applicability.status` is `not_applicable` (`advisoryWait.convergenceScope: "idd-claimed"` and this PR has no matching verified linked claim/branch).   |
+| `review-pending`                          | The primary bot has not yet reviewed current HEAD (`pending: true`). Always co-occurs with `review-item-count-unknown` below, since an off-HEAD review reports no usable item count. |
+| `unresolved-copilot-threads`              | `threads.satisfied` is `false` -- at least one Copilot-authored thread is neither resolved nor validly dispositioned.                                   |
+| `missing-regular-comment-disposition`     | `dispositionEvidence.missingRegularCommentCount` is non-zero -- an outstanding regular (non-thread) PR comment still lacks a fresh disposition marker.  |
+| `review-item-count-unknown`               | The latest review is on current HEAD but its comment count is unavailable (a GraphQL nullable-field edge case).                                         |
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply idd-template/docs/idd-helper-scripts.md]: **Accepted** — fixed in fc443d1aaaa32414bf6df312d400613dfd0db53a: same fix applied to this file's mirrored copy — corrected the `review-item-count-unknown` row to also describe the off-HEAD case (co-fires with `review-pending`), not only the on-HEAD nullable-count edge case.

**Comment by kurone-kito** [comment]
**Rejected** — chatgpt-codex-connector[bot] did not review HEAD fc443d1aaaa32414bf6df312d400613dfd0db53a (Codex usage limits for code reviews reached); this is not a completed review (source: #issuecomment-5148930472)

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD fc443d1aaaa32414bf6df312d400613dfd0db53a (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5148930613)

**Comment by kurone-kito** [comment]
_claude-code-8e600fc12aaea6bb: review triage snapshot — IDD automation marker. Do not edit._

**Comment by kurone-kito** [comment]
| Field | Value |
| --- | --- |
| Phase | F3 merged |
| Claim | cc-20260801T011714-8dcafae7 |
| Branch | issue/1719-feat-advisory-convergence-explain-why |
| Last checked | 2026-08-01T02:24:51.771Z |
| Open blockers | none |
| Next action | F4 cleanup then F5 discover |
| Authoritative by | merge commit f515fc34befc9026bdce4686fc6e41cc90e15c1d, matched head 9c42f4c92e9ecc400bc29a9df5a1ca3971206a89 |
